presto-parser/src/main/java/io/prestosql/sql/tree/Limit.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+public class Limit
+        extends Node
+{
+    private final Optional<String> limit;
```

Comment:
`limit` should never be `empty()`, right? That would syntactically invalid.

Reply:
Yes, and at first I thought of `private final String limit;`, but I decided to use `Optional`, because originally there was `Optional<String> limit` field in `Query` and `QuerySpecification`.
Shall I add a check in the constructor?

Reply:
Let's just make it a `private final String limit;`. It was `Optional<String>` in `Query`/`QuerySpecification` to indicate that the clause could be missing. If there's a `Limit` node, the clause is *present*, so no need to have another level of `Optional` that will never be `empty`. The constructor should not take an `Optional`, either. It should be up to the caller to pass in the required value.